### ISSUE TITLE
Upload problem

### ISSUE THREAD

**Issue by it-can** [fuel/core#2026]
I updated a fuelphp 1.5 app to the latest 1.9 develop, everything seems to work except uploading files... I get this error:

```
Undefined offset: 1

COREPATH/classes/input/instance.php @ line 434
```

this is my form:

```
<?php echo Form::open(array('class'=>'well', 'enctype'=>'multipart/form-data')); ?>
<?php echo Form::input('subject', $subject); ?>

<?php echo Form::file('file[]'); ?>
<?php echo Form::file('file[]'); ?>
<?php echo Form::file('file[]'); ?>
<?php echo Form::file('file[]'); ?>
<?php echo Form::file('file[]'); ?>

<?php echo Form::submit('save', 'Upload'); ?>
<?php echo Form::close(); ?>
```

in my controller:

```
$config = array(
    'path'  => DOCROOT . 'temp',
    'field' => 'file',
);

$files = array();

// process the uploaded files in $_FILES
Upload::process($config);

// if there are any valid files
if (Upload::is_valid())
{
    // save them according to the config
    Upload::save();

    foreach (Upload::get_files() AS $f)
    {
        $file    = $f['saved_to'] . $f['saved_as'];
        $files[] = $file;
    }
}
```

**Comment by it-can** [comment]
Also happens on a other project that is on 1.9/develop, code seems to fail since commit 90c357e93911278c610c3c7f7aca7156aebab359

**Comment by WanWizard** [comment]
Yeah, was a bit to quick with the input fix, I'm busy reworking it.

**Comment by WanWizard** [comment]
Just pushed.

**Comment by WanWizard** [comment]
I assume it fixed it?

**Comment by it-can** [comment]
Seems to work fine

**Comment by it-can** [comment]
Isnt there a way to unit test this?

**Comment by WanWizard** [comment]
Difficult. The problem is that there are a lot of ways to present input. I bumped into the bug in `hydrate()` when an app was using PUT and DELETE, but the input remained empty. Turned out the processing of the different content types was all wrong, and some input was just silenty dropped. So I added the DomainException you bumped into. When doing so I missed the fact that with multipart/form-data, the data is actually part of the content type so that got stripped out by mistake. 

I have test controllers to test different types of input, including file uploads, but the truth is I just didn't run them this time.... my bad. :-(
